app/helpers/application_helper.rb

```diff
@@ -18,6 +18,14 @@ def state(document)
 
   end
 
+  def show_preview?(item)
+    draft = item.draft?
+    if item.respond_to?(:documents)
+      draft ||= item.documents.any?(&:draft?)
+    end
+    draft
+  end
+
```

Comment:
This looks like it'll work, but would confuse me less like this:

```
def show_preview?(item)
  return item.draft? unless item.respond_to?(:documents)
  
  item.draft? || item.documents.any?(&:draft?)
end
```
or this:
```
def show_preview?(item)
  if item.respond_to?(:documents)
    item.draft? || item.documents.any?(&:draft?)
  else
    item.draft?
  end
end
```

What do you think?